TypeError: Cannot read property 'postgres' of undefined

Hey there! I'm working on getting this installed on Ubuntu 14.04 and having trouble with the `readme_server.js` example. It seems that it's trying to read from the `global.environment` object without actually having that variable set anywhere. Here's the error I'm getting:

```console
vagrant@vagrant-ubuntu-trusty-64:~/Windshaft$ node examples/readme_server.js

/home/vagrant/Windshaft/lib/windshaft/renderers/mapnik/factory.js:12
var COLUMN_NAME_DEFAULT = global.environment.postgres.geometry_field || 'the_g
                                            ^
TypeError: Cannot read property 'postgres' of undefined
    at Object.<anonymous> (/home/vagrant/Windshaft/lib/windshaft/renderers/mapnik/factory.js:12:45)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/vagrant/Windshaft/lib/windshaft/renderers/mapnik/index.js:2:26)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
```

I've looked in both `lib/windshaft/renderers/mapnik/index.js` and `lib/windshaft/renderers/mapnik/factory.js` but can't find where `global.environment` would've been defined, leading me to believe that this may be an actual bug rather than me being dense?

For your reference, here's my log of my successful `npm install` using Windshaft v0.38.2.

```console
vagrant@vagrant-ubuntu-trusty-64:~/Windshaft$ node --version
v0.10.36
vagrant@vagrant-ubuntu-trusty-64:~/Windshaft$ git checkout -b 0.38.2 tags/0.38.2
Switched to a new branch '0.38.2'
vagrant@vagrant-ubuntu-trusty-64:~/Windshaft$ npm install

> windshaft@0.38.2 preinstall /home/vagrant/Windshaft
> sh ./scripts/check-node-canvas.sh

./scripts/check-node-canvas.sh: 3: ./scripts/check-node-canvas.sh: [[: not found

> canvas@1.2.1 install /home/vagrant/Windshaft/node_modules/canvas
> node-gyp rebuild

make: Entering directory `/home/vagrant/Windshaft/node_modules/canvas/build'
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node
  SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node: Finished
  COPY Release/canvas-postbuild.node
  CXX(target) Release/obj.target/canvas/src/Canvas.o
  CXX(target) Release/obj.target/canvas/src/CanvasGradient.o
  CXX(target) Release/obj.target/canvas/src/CanvasPattern.o
  CXX(target) Release/obj.target/canvas/src/CanvasRenderingContext2d.o
../src/CanvasRenderingContext2d.cc: In member function ‘void Context2d::shadow(void (*)(cairo_t*))’:
../src/CanvasRenderingContext2d.cc:374:22: warning: unused variable ‘surface’ [-Wunused-variable]
     cairo_surface_t *surface = cairo_get_group_target(_context);
                      ^
  CXX(target) Release/obj.target/canvas/src/color.o
  CXX(target) Release/obj.target/canvas/src/Image.o
  CXX(target) Release/obj.target/canvas/src/ImageData.o
  CXX(target) Release/obj.target/canvas/src/init.o
  CXX(target) Release/obj.target/canvas/src/PixelArray.o
  CXX(target) Release/obj.target/canvas/src/FontFace.o
  SOLINK_MODULE(target) Release/obj.target/canvas.node
  SOLINK_MODULE(target) Release/obj.target/canvas.node: Finished
  COPY Release/canvas.node
make: Leaving directory `/home/vagrant/Windshaft/node_modules/canvas/build'

> hiredis@0.1.17 install /home/vagrant/Windshaft/node_modules/redis-mpool/node_modules/hiredis
> node-gyp rebuild

make: Entering directory `/home/vagrant/Windshaft/node_modules/redis-mpool/node_modules/hiredis/build'
  CC(target) Release/obj.target/hiredis/deps/hiredis/hiredis.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/net.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/sds.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/async.o
  AR(target) Release/obj.target/deps/hiredis.a
  COPY Release/hiredis.a
  CXX(target) Release/obj.target/hiredis/src/hiredis.o
  CXX(target) Release/obj.target/hiredis/src/reader.o
  SOLINK_MODULE(target) Release/obj.target/hiredis.node
  SOLINK_MODULE(target) Release/obj.target/hiredis.node: Finished
  COPY Release/hiredis.node
make: Leaving directory `/home/vagrant/Windshaft/node_modules/redis-mpool/node_modules/hiredis/build'

> pg@2.6.2-cdb1 install /home/vagrant/Windshaft/node_modules/cartodb-psql/node_modules/pg
> node-gyp rebuild || (exit 0)

make: Entering directory `/home/vagrant/Windshaft/node_modules/cartodb-psql/node_modules/pg/build'
  CXX(target) Release/obj.target/binding/src/binding.o
  SOLINK_MODULE(target) Release/obj.target/binding.node
  SOLINK_MODULE(target) Release/obj.target/binding.node: Finished
  COPY Release/binding.node
make: Leaving directory `/home/vagrant/Windshaft/node_modules/cartodb-psql/node_modules/pg/build'

> hiredis@0.1.17 install /home/vagrant/Windshaft/node_modules/grainstore/node_modules/redis-mpool/node_modules/hiredis
> node-gyp rebuild

make: Entering directory `/home/vagrant/Windshaft/node_modules/grainstore/node_modules/redis-mpool/node_modules/hiredis/build'
  CC(target) Release/obj.target/hiredis/deps/hiredis/hiredis.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/net.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/sds.o
  CC(target) Release/obj.target/hiredis/deps/hiredis/async.o
  AR(target) Release/obj.target/deps/hiredis.a
  COPY Release/hiredis.a
  CXX(target) Release/obj.target/hiredis/src/hiredis.o
  CXX(target) Release/obj.target/hiredis/src/reader.o
  SOLINK_MODULE(target) Release/obj.target/hiredis.node
  SOLINK_MODULE(target) Release/obj.target/hiredis.node: Finished
  COPY Release/hiredis.node
make: Leaving directory `/home/vagrant/Windshaft/node_modules/grainstore/node_modules/redis-mpool/node_modules/hiredis/build'
npm WARN deprecated set-immediate@0.1.1: Use `setimmediate` instead

> mapnik@1.4.15-cdb1 install /home/vagrant/Windshaft/node_modules/mapnik
> node-pre-gyp install --fallback-to-build

[mapnik] Success: "/home/vagrant/Windshaft/node_modules/mapnik/lib/binding/node-v11-linux-x64/mapnik.node" is installed via remote

> sqlite3@2.2.7 install /home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

[sqlite3] Success: "/home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/sqlite3/lib/binding/node-v11-linux-x64/node_sqlite3.node" is installed via remote

> zipfile@0.5.7 install /home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/zipfile
> node-pre-gyp install --fallback-to-build

[zipfile] Success: "/home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/zipfile/lib/binding/node-v11-linux-x64/zipfile.node" is installed via remote

> srs@0.4.6 install /home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/srs
> node-pre-gyp install --fallback-to-build

[srs] Success: "/home/vagrant/Windshaft/node_modules/grainstore/node_modules/millstone/node_modules/srs/lib/binding/node-v11-linux-x64/srs.node" is installed via remote
sphericalmercator@1.0.2 node_modules/sphericalmercator

step-profiler@0.1.0 node_modules/step-profiler

queue-async@1.0.7 node_modules/queue-async

step@0.0.5 node_modules/step

semver@1.1.4 node_modules/semver

node-statsd@0.0.7 node_modules/node-statsd

chronograph@0.1.0 node_modules/chronograph

underscore@1.6.0 node_modules/underscore

redis@0.8.6 node_modules/redis

mocha@1.21.5 node_modules/mocha
├── escape-string-regexp@1.0.2
├── diff@1.0.8
├── growl@1.8.1
├── commander@2.3.0
├── debug@2.0.0 (ms@0.6.2)
├── mkdirp@0.5.0 (minimist@0.0.8)
├── glob@3.2.3 (inherits@2.0.1, graceful-fs@2.0.3, minimatch@0.2.14)
└── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)

request@2.48.0 node_modules/request
├── caseless@0.7.0
├── json-stringify-safe@5.0.0
├── forever-agent@0.5.2
├── aws-sign2@0.5.0
├── stringstream@0.0.4
├── oauth-sign@0.5.0
├── tunnel-agent@0.4.0
├── node-uuid@1.4.2
├── qs@2.3.3
├── mime-types@1.0.2
├── combined-stream@0.0.7 (delayed-stream@0.0.5)
├── form-data@0.1.4 (mime@1.2.11, async@0.9.0)
├── http-signature@0.10.1 (assert-plus@0.1.5, asn1@0.1.11, ctype@0.5.3)
├── bl@0.9.4 (readable-stream@1.0.33)
├── tough-cookie@0.12.1 (punycode@1.3.2)
└── hawk@1.1.1 (cryptiles@0.2.2, sntp@0.2.4, boom@0.4.2, hoek@0.9.1)

express@2.5.11 node_modules/express
├── qs@0.4.2
├── mime@1.2.4
├── mkdirp@0.3.0
└── connect@1.9.2 (formidable@1.0.17)

tilelive-mapnik@0.6.12 node_modules/tilelive-mapnik
├── generic-pool@2.1.1
└── mime@1.2.11

tilelive@4.5.3 node_modules/tilelive
└── optimist@0.6.1 (wordwrap@0.0.2, minimist@0.0.10)

jshint@2.6.3 node_modules/jshint
├── strip-json-comments@1.0.2
├── exit@0.1.2
├── minimatch@1.0.0 (sigmund@1.0.0, lru-cache@2.5.0)
├── console-browserify@1.1.0 (date-now@0.1.4)
├── shelljs@0.3.0
├── cli@0.6.5 (glob@3.2.11)
└── htmlparser2@3.8.2 (domelementtype@1.3.0, entities@1.0.0, domhandler@2.3.0, readable-stream@1.1.13, domutils@1.5.1)

carto@0.15.1-cdb1 node_modules/carto
├── optimist@0.6.1 (wordwrap@0.0.2, minimist@0.0.10)
└── mapnik-reference@6.0.5

torque.js@2.10.0 node_modules/torque.js

abaculus@1.1.0 node_modules/abaculus

canvas@1.2.1 node_modules/canvas
└── nan@1.5.3

redis-mpool@0.3.0 node_modules/redis-mpool
├── generic-pool@2.1.1
├── redis@0.12.1
└── hiredis@0.1.17 (bindings@1.2.1, nan@1.1.2)

cartodb-psql@0.4.0 node_modules/cartodb-psql
└── pg@2.6.2-cdb1 (buffer-writer@1.0.0, generic-pool@2.0.3)

mapnik@1.4.15-cdb1 node_modules/mapnik
├── nan@1.2.0
├── mapnik-vector-tile@0.5.5
└── node-pre-gyp@0.5.25

grainstore@0.23.0 node_modules/grainstore
├── carto@0.9.5-cdb2 (underscore@1.4.4, optimist@0.6.1, mapnik-reference@5.0.9, xml2js@0.2.8)
├── redis-mpool@0.1.0 (generic-pool@2.1.1, redis@0.12.1, hiredis@0.1.17)
└── millstone@0.6.14 (generic-pool@2.0.4, mime@1.2.11, mkdirp@0.3.5, optimist@0.6.1, request@2.34.0, sqlite3@2.2.7, zipfile@0.5.7, srs@0.4.6)
```

Thanks for your help!

--- a/examples/readme_server.js
+++ b/examples/readme_server.js
@@ -5,6 +5,15 @@
 
 var Windshaft = require('../lib/windshaft');
 var _         = require('underscore');
+
+// Force 'test' environment
+var ENV = 'test';
+
+// set environment specific variables
+global.settings     = require('../config/settings');
+global.environment  = require('../config/environments/' + ENV);
+_.extend(global.settings, global.environment);
+
 var config = {
     base_url: '/database/:dbname/table/:table',
     base_url_notable: '/database/:dbname',

